src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -725,7 +725,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
 
     def setFlag(mask: Long): this.type   = { _rawflags |= mask ; this }
     def resetFlag(mask: Long): this.type = { _rawflags &= ~mask ; this }
-    def resetFlags() { rawflags &= TopLevelCreationFlags }
+    def resetFlags() { rawflags = 0 }
```

Comment:
the motivation was indeed `final`, see the test case: https://github.com/scala/scala/pull/5202/commits/fbf549b615786f5d749de5022b0fefdc1e2ab435#diff-80a0746bc22d5376fd194dfb8985330d

i think the non-hacky solution would be to remove `TopLevelCreationFlags`, its uses are more like "oh, this seems to be a well-fitting collection of flags for my use-case"